lib/datadog/core/environment/variable_helpers.rb

```diff
@@ -98,6 +98,53 @@ def env_to_list(var, default = [], comma_separated_only:, deprecation_warning: t
           end
         end
 
+        # Parses comma- or space-separated lists of key:value entries
+        #
+        # If a comma is present, then the list is considered comma-separated.
+        # Otherwise, it is considered space-separated.
+        #
+        # After the entries are separated, commas and whitespaces that are
+        # either trailing or leading are trimmed.
+        #
+        # Each entry is then separated by ':' into an entry of a Hash{key => value}
+        #
+        # Empty entries, after trimmed, are also removed from the result.
+        #
+        #
+        # @param [String] var environment variable
+        # @param [Array<String>] var list of environment variables
+        # @param [Array<Object>] default the default value if the keys in `var` are not present in the environment
+        # @param [Boolean] deprecation_warning when `var` is a list, record a deprecation log when
+        #   the first key in `var` is not used.
+        # @return [Hash<String>] if the environment value is a valid list of key:value entries
+        # @return [default] if the environment value is not found
+        def env_to_hash(var, default = [], comma_separated_only:, deprecation_warning: true)
```

Comment:
Thanks, @ivoanjo, for the call. 

Yes, @zarirhamza, I think having a way to convert ENV variable values into a hash is a great idea. Still , since we want to deprecate this file, let's wait for this to get merged https://github.com/DataDog/dd-trace-rb/pull/2983, and add that functionality there

Reply:
Addressed above